pkg/controller/operators/catalog/operator.go

```diff
@@ -1423,6 +1444,20 @@ func (o *Operator) syncInstallPlans(obj interface{}) (syncError error) {
 	return
 }
 
+func (o *Operator) requeueSubscriptionForInstallPlan(plan *v1alpha1.InstallPlan, logger *logrus.Entry) {
+	// Notify subscription loop of installplan changes
+	if owners := ownerutil.GetOwnersByKind(plan, v1alpha1.SubscriptionKind); len(owners) > 0 {
+		for _, owner := range owners {
+			logger.WithField("owner", owner).Debug("requeueing installplan owner")
+			if err := o.subQueueSet.Requeue(plan.GetNamespace(), owner.Name); err != nil {
+				logger.WithError(err).Warn("error requeuing installplan owner")
+			}
+		}
+		return
+	}
+	logger.Trace("no installplan owner subscriptions found to requeue")
```

Comment:
(the actual implementation seems reasonable to me)

Minor nit: I think we can also just reverse the conditional check to be `if owners := ...; len(owners) <= 0` and just move that log message and return early in that conditional, which allows us to move that for loop further to the right for better readability.

Reply:
Yeah I just moved this block of code as is to reuse it, but I can refactor it to reduce nesting as well.